cmd/serv.go

```diff
@@ -205,7 +202,7 @@ func runServ(c *cli.Context) error {
 	username := strings.ToLower(rr[0])
 	reponame := strings.ToLower(strings.TrimSuffix(rr[1], ".git"))
 
-	if alphaDashDotPattern.MatchString(reponame) {
+	if !user_model.IsValidUsername(reponame) {
```

Comment:
this is `reponame`, maybe you mean `username`?

Reply:
Before they all used the same `alphaDashDotPattern` rule.

So maybe it's not bad to keep the same rule for username/reponame (since the PR was WIP and was expected to reduce unnecessary changes, so the function was reused, indeed it's not ideal enough).